src/params/types.ts

```diff
@@ -293,12 +293,18 @@ export interface SelectOptions<T = unknown> {
   value: T;
 }
 
+/**
```

Comment:
Can we add a comment here with a brief explanation of why `NoInfer` is needed here for future maintainers?